src/compiler/checker.ts

```diff
@@ -1192,6 +1195,40 @@ namespace ts {
                     return false;
                 });
             }
+
+            function isPropertyImmediatelyReferencedWithinDeclaration(declaration: PropertyDeclaration, usage: Node) {
+                // always legal if usage is after declaration
+                if (usage.end > declaration.end) {

```

Comment:
same applies if usage is before declaration

Reply:
That other error is completely unrelated to the purpose of this function. Looking at the function name alone makes it clear that it should only handle uses within the range of the declaration.

Reply:
Agreed in principle, but I'm not sure if you're asking for a change here? 🙂